src/mongo_logger.ts

```diff
@@ -122,58 +178,262 @@ function resolveLogPath(
+            ...log,
+            maxIdleTimeMS,
+            minPoolSize,
+            maxPoolSize,
+            maxConnecting,
+            waitQueueTimeoutMS
+          };
+          log.waitQueueSize = ev.waitQueueSize;
+        }
+        break;
+      case CONNECTION_POOL_READY:
+        ev = logObject as ConnectionPoolReadyEvent;
+        log.message = 'Connection pool ready';
+        break;
+      case CONNECTION_POOL_CLEARED:
+        ev = logObject as ConnectionPoolClearedEvent;
+        log.message = 'Connection pool cleared';
+        if (ev.serviceId?._bsontype === 'ObjectId') {
+          log.serviceId = ev.serviceId.toHexString();
+        }
+        break;
+      case CONNECTION_POOL_CLOSED:
+        ev = logObject as ConnectionPoolClosedEvent;
+        log.message = 'Connection pool closed';
+        break;
+      case CONNECTION_CREATED:
+        ev = logObject as ConnectionCreatedEvent;
+        log.message = 'Connection created';
+        log.driverConnectionId = ev.connectionId;
+        break;
+      case CONNECTION_READY:
+        ev = logObject as ConnectionReadyEvent;
+        log.message = 'Connection ready';
+        log.driverConnectionId = ev.connectionId;
+        break;
+      case CONNECTION_CLOSED:
+        ev = logObject as ConnectionClosedEvent;
+        log.message = 'Connection closed';
+        log.driverConnectionId = ev.connectionId;
+        switch (ev.reason) {
+          case 'stale':
+            log.reason = 'Connection became stale because the pool was cleared';
+            break;
+          case 'idle':
+            log.reason =
+              'Connection has been available but unused for longer than the configured max idle time';
+            break;
+          case 'error':
+            log.reason = 'An error occurred while using the connection';
+            if (ev.error) {
+              log.error = ev.error;
+            }
+            break;
+          case 'poolClosed':
+            log.reason = 'Connection pool was closed';
+            break;
+          default:
+          // Omit if we have some other reason as it would be invalid
+        }
+        break;
+      case CONNECTION_CHECK_OUT_STARTED:
+        ev = logObject as ConnectionCheckOutStartedEvent;
+        log.message = 'Connection checkout started';
+        break;
+      case CONNECTION_CHECK_OUT_FAILED:
+        ev = logObject as ConnectionCheckOutFailedEvent;
+        log.message = 'Connection checkout failed';
+        log.reason = ev.reason;
+        break;
+      case CONNECTION_CHECKED_OUT:
+        ev = logObject as ConnectionCheckedOutEvent;
+        log.message = 'Connection checked out';
+        log.driverConnectionId = ev.connectionId;
+        break;
+      case CONNECTION_CHECKED_IN:
+        ev = logObject as ConnectionCheckedInEvent;
+        log.message = 'Connection checked in';
+        log.driverConnectionId = ev.connectionId;
+        break;
+    }
+  } else {
+    for (const key in logObject) {
```

Comment:
Hm for...in is not typically what we would use b/c of enumerating keys from the prototype chain, but I suppose that is precisely what we want b/c ConnectionPoolMonitoringEvent uses inheritance. So just making sure that is intentional?

Reply:
The intention here was just to pass through an arbitrary object that isn't a command event or a CMAP event excepting the top-level nullish keys. Would it be more appropriate to use `Object.entries` here?

Reply:
Yep that works! Oh yes I see now that the CMAP event is handled earlier so inheritance isn't an issue here.